service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java

```diff
@@ -44,9 +42,16 @@
     private ExecutorService executor;
     private IotHubConnectionString iotHubConnectionString;
 
-    @Getter
-    @Setter
-    private Proxy proxy;
+    private RegistryManagerOptions options;
+
+    /**
+     * As of release 1.22.0, replaced by {@link #createFromConnectionString()}
+     */
+    @Deprecated
+    public RegistryManager()
```

Comment:
what is this for? did we allow creation of registry manager without having to pass in even a connection string?

Reply:
Java has a funny quirk where a class with no public constuctor automatically has a public constructor. This commit just makes that constuctor visible. I believe it was an accident that we didn't make the only constuctor protected

Reply:
Ah, the default constructor, yeah. Can we not make it private now? Also, can we modify the comment to say that this is simply the default java constructor. the `replaced` text makes it seem that we did allow a no args initialization of registry manager.

Reply:
Since it was a public constructor, we can't just make it private now. That would be a breaking change. We did technically allow users to build a registry manager through that constructor. I'll add to the header here that it was once the default constructor, though

Reply:
Oh, I am confused, if this worked, why are we deprecating this now? It is because it doesn't really do anything useful?